### ISSUE TITLE
mlogin could disable Manta's abort-on-core behavior

### ISSUE THREAD

**Issue by davepacheco** [joyent/node-manta#244]
Manta job tasks currently fail immediately when any process dumps core:

```
https://apidocs.joyent.com/manta/jobs-reference.html#task-lifetime-success-and-failure-exec-property
```

I think that makes sense for most jobs, but probably not for interactive jobs.  mlogin could work around this by running child processes in a contract using "ctrun -i signal,core".

**Comment by davepacheco** [comment]
Here's an example of what happens before this issue is fixed:

```
$ ./bin/mlogin 
 * created interactive job -- 703538b2-1748-c04d-b946-e9032df29a00
 * waiting for session...   established
poseidon@manta # node -e 'process.abort()'


 * session complete
 * cleaning up resources...
```

Here's what happens after the change:

```
$ ./bin/mlogin 
 * created interactive job -- 761a193f-b4bb-e1f2-aee8-ddaf233ff4e6
 * waiting for session...   established
poseidon@manta # ptree $$
16511 zsched
  18276 /sbin/init
    20914 /opt/marlin/build/node/bin/node --abort-on-uncaught-exception /opt/marlin/
      95309 ./node lib/agent.js
        95341 /usr/bin/ctrun -i core -l child /bin/bash --norc
          95343 /bin/bash --norc
            95370 ptree 95343
poseidon@manta # node -e 'process.abort()'
Abort (core dumped)
poseidon@manta # ls /var/tmp/core
mrcore.node.95782
poseidon@manta # ls -l /var/tmp/core
total 42139
-rw------- 1 root root 43000741 Jul 21 19:09 mrcore.node.95782
poseidon@manta # exit


 * remote process exited
 * cleaning up resources...
 * session complete
dap@b44c74d6 node-manta-244 $
```

The `-c` option also still works:

```
$ ./bin/mlogin -c date
 * created interactive job -- f5c0d8fb-8d4b-c7fb-acdf-e404e94616d5
 * waiting for session...   established
Fri Jul 21 19:08:48  2017


 * remote process exited
 * cleaning up resources...
 * session complete
```

The change is `make prepush` clean except for the MPU tests:

```
$ make prepush
version is: 4.4.2
[[ `cat package.json | json version` == `grep '^## ' CHANGES.md | head -2 | tail -1 | awk '{print $2}'` ]]
deps/javascriptlint/build/install/jsl --nologo --nosummary --conf=tools/jsl.node.conf lib/jobshare.js lib/streaming_json_stream.js lib/trackmarker.js lib/string_stream.js lib/utils.js lib/create_client.js lib/queue.js lib/options.js lib/index.js lib/client.js test/mput.test.js test/mmpu.test.js test/utils.test.js test/trackmarker.test.js test/lib/logging.js test/muntar.test.js test/mjob-simple.test.js test/completion.test.js test/mget.test.js test/mfind.test.js test/client.test.js bin/mmkdir bin/mchattr bin/mjob bin/mrm bin/msign bin/mln bin/mfind bin/muntar bin/mmd5 bin/mls bin/mlogin bin/mput bin/mrmdir bin/minfo bin/mmpu bin/mchmod bin/mget
/home/dap/node-manta-244/lib/jobshare.js
/home/dap/node-manta-244/lib/streaming_json_stream.js
/home/dap/node-manta-244/lib/trackmarker.js
/home/dap/node-manta-244/lib/string_stream.js
/home/dap/node-manta-244/lib/utils.js
/home/dap/node-manta-244/lib/create_client.js
/home/dap/node-manta-244/lib/queue.js
/home/dap/node-manta-244/lib/options.js
/home/dap/node-manta-244/lib/index.js
/home/dap/node-manta-244/lib/client.js
/home/dap/node-manta-244/test/mput.test.js
/home/dap/node-manta-244/test/mmpu.test.js
/home/dap/node-manta-244/test/utils.test.js
/home/dap/node-manta-244/test/trackmarker.test.js
/home/dap/node-manta-244/test/lib/logging.js
/home/dap/node-manta-244/test/muntar.test.js
/home/dap/node-manta-244/test/mjob-simple.test.js
/home/dap/node-manta-244/test/completion.test.js
/home/dap/node-manta-244/test/mget.test.js
/home/dap/node-manta-244/test/mfind.test.js
/home/dap/node-manta-244/test/client.test.js
/home/dap/node-manta-244/bin/mmkdir
/home/dap/node-manta-244/bin/mchattr
/home/dap/node-manta-244/bin/mjob
/home/dap/node-manta-244/bin/mrm
/home/dap/node-manta-244/bin/msign
/home/dap/node-manta-244/bin/mln
/home/dap/node-manta-244/bin/mfind
/home/dap/node-manta-244/bin/muntar
/home/dap/node-manta-244/bin/mmd5
/home/dap/node-manta-244/bin/mls
/home/dap/node-manta-244/bin/mlogin
/home/dap/node-manta-244/bin/mput
/home/dap/node-manta-244/bin/mrmdir
/home/dap/node-manta-244/bin/minfo
/home/dap/node-manta-244/bin/mmpu
/home/dap/node-manta-244/bin/mchmod
/home/dap/node-manta-244/bin/mget
deps/jsstyle/jsstyle -f tools/jsstyle.conf lib/jobshare.js lib/streaming_json_stream.js lib/trackmarker.js lib/string_stream.js lib/utils.js lib/create_client.js lib/queue.js lib/options.js lib/index.js lib/client.js test/mput.test.js test/mmpu.test.js test/utils.test.js test/trackmarker.test.js test/lib/logging.js test/muntar.test.js test/mjob-simple.test.js test/completion.test.js test/mget.test.js test/mfind.test.js test/client.test.js bin/mmkdir bin/mchattr bin/mjob bin/mrm bin/msign bin/mln bin/mfind bin/muntar bin/mmd5 bin/mls bin/mlogin bin/mput bin/mrmdir bin/minfo bin/mmpu bin/mchmod bin/mget
check ok
npm install
unset MANTA_DEFAULT_CONTENT_TYPE; \
        if [[ -z "" ]]; then \
                ./node_modules/.bin/nodeunit test/*.test.js; \
        else \
                echo "# Running subset of tests matching TEST_FILTER="; \
                ./node_modules/.bin/nodeunit  test/client.test.js test/completion.test.js test/mfind.test.js test/mget.test.js test/mjob-simple.test.js test/mmpu.test.js test/mput.test.js test/muntar.test.js test/trackmarker.test.js test/utils.test.js; \
        fi

client.test.js
✔ mkdir
✔ mkdir (sub)
✔ put
✔ #231: put (special characters)
✔ #231: ls (special characters)
✔ #231: get (special characters)
✔ #231: rm (special characters)
✔ chattr
✔ put (zero byte streaming)
✔ put without mkdirp
✔ put with mkdirp
✔ streams
✔ put MD5 mismatch
✔ GH-72 content-length: undefined
✔ ls
✔ createListStream
✔ createListStream (dir only)
✔ createListStream (object only)
✔ ln
✔ info (link)
✔ ftw
✔ create job (simple grep)
✔ get job
✔ add input keys
✔ get job input
✔ end job
✔ wait for job
✔ get job output
✔ create and cancel job
✔ unlink object
✔ unlink link
✔ rmr
✔ mkdirp/rmr
✖ create upload

ResourceNotFoundError: /poseidon/uploads does not exist
    at parseResponse (/home/dap/node-manta-244/node_modules/restify-clients/lib/JsonClient.js:73:26)
    at IncomingMessage.done (/home/dap/node-manta-244/node_modules/restify-clients/lib/StringClient.js:207:13)
    at IncomingMessage.g (events.js:180:16)
    at IncomingMessage.emit (events.js:117:20)
    at _stream_readable.js:944:16
    at process._tickCallback (node.js:458:13)

✖ upload part

AssertionError: 'undefined' === 'string'
    at MantaClient.uploadPart (/home/dap/node-manta-244/lib/client.js:3396:12)
    at Object.opts.account (/home/dap/node-manta-244/test/client.test.js:761:17)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:232:20
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:168:13
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:131:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:165:17
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:463:34
    at createClient (/home/dap/node-manta-244/test/client.test.js:74:9)
    at Object.module.exports.setUp (/home/dap/node-manta-244/test/client.test.js:78:9)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:260:35
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:458:21
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:163:13
    at iterate (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:123:13)
    at async.forEachSeries (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:139:9)
    at _asyncMap (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:162:9)
    at Object.mapSeries (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:152:23)

✖ get upload

AssertionError: 'undefined' === 'string'
    at MantaClient.getUpload (/home/dap/node-manta-244/lib/client.js:3516:12)
    at Object.opts.account (/home/dap/node-manta-244/test/client.test.js:785:17)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:232:20
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:168:13
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:131:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:165:17
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:463:34
    at createClient (/home/dap/node-manta-244/test/client.test.js:74:9)
    at Object.module.exports.setUp (/home/dap/node-manta-244/test/client.test.js:78:9)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:260:35
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:458:21
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:163:13
    at iterate (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:123:13)
    at async.forEachSeries (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:139:9)
    at _asyncMap (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:162:9)
    at Object.mapSeries (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:152:23)

✖ commit upload

AssertionError: 'undefined' === 'string'
    at MantaClient.commitUpload (/home/dap/node-manta-244/lib/client.js:3567:12)
    at Object.opts.account (/home/dap/node-manta-244/test/client.test.js:805:17)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:232:20
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:168:13
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:131:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:165:17
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:463:34
    at createClient (/home/dap/node-manta-244/test/client.test.js:74:9)
    at Object.module.exports.setUp (/home/dap/node-manta-244/test/client.test.js:78:9)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:260:35
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:458:21
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:163:13
    at iterate (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:123:13)
    at async.forEachSeries (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:139:9)
    at _asyncMap (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:162:9)
    at Object.mapSeries (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:152:23)

✖ abort upload

ResourceNotFoundError: /poseidon/uploads does not exist
    at parseResponse (/home/dap/node-manta-244/node_modules/restify-clients/lib/JsonClient.js:73:26)
    at IncomingMessage.done (/home/dap/node-manta-244/node_modules/restify-clients/lib/StringClient.js:207:13)
    at IncomingMessage.g (events.js:180:16)
    at IncomingMessage.emit (events.js:117:20)
    at _stream_readable.js:944:16
    at process._tickCallback (node.js:458:13)

✖ #311: create upload with special headers

ResourceNotFoundError: /poseidon/uploads does not exist
    at parseResponse (/home/dap/node-manta-244/node_modules/restify-clients/lib/JsonClient.js:73:26)
    at IncomingMessage.done (/home/dap/node-manta-244/node_modules/restify-clients/lib/StringClient.js:207:13)
    at IncomingMessage.g (events.js:180:16)
    at IncomingMessage.emit (events.js:117:20)
    at _stream_readable.js:944:16
    at process._tickCallback (node.js:458:13)

✔ GH-196 getPath ~~/
✔ #180: Invalid key results in no client error
✔ MANTA-2812 null signer
✔ MANTA-2812 undefined signer

completion.test.js
✔ mchattr --completion
✔ mchmod --completion
✔ mfind --completion
✔ mget --completion
✔ minfo --completion
✔ mjob --completion
✔ mln --completion
✔ mlogin --completion
✔ mls --completion
✔ mmd5 --completion
✔ mmkdir --completion
✔ mmpu --completion
✔ mput --completion
✔ mrm --completion
✔ mrmdir --completion
✔ msign --completion
✔ muntar --completion

mfind.test.js
✔ setup: create test tree at /poseidon/stor/node-manta-test-mfind-aa5aa1a1
✔ check if operator (mfind forbidden)
✔ mfind TESTDIR
✔ mfind -j TESTDIR
✔ mfind TESTDIR/afile.txt
✔ mfind TESTDIR/notafile.txt
✔ mfind TESTDIR/notafile.txt TESTDIR/afile.txt
WARNING: skipping mfind forbidden test: user is an operator
✔ mfind /poseidon/stor TESTDIR/afile.txt
✔ cleanup: rm test tree /poseidon/stor/node-manta-test-mfind-aa5aa1a1

mget.test.js
✔ setup: create test tree at /poseidon/stor/node-manta-test-mget-129d3802
✔ mget TESTDIR/02.data
✔ mget TESTDIR/01.data TESTDIR/02.data TESTDIR/03.data
✔ mget TESTDIR/01.txt TESTDIR/02.txt TESTDIR/03.txt
✔ mget -o TMPFILE TESTDIR/01.txt TESTDIR/02.txt TESTDIR/03.txt
✔ mget TESTDIR/01.txt TESTDIR/02.txt TESTDIR/XX.txt TESTDIR/03.txt
✔ mget -o TMPFILE TESTDIR/01.txt TESTDIR/02.txt TESTDIR/XX.txt TESTDIR/03.txt
✔ cleanup: rm test tree /poseidon/stor/node-manta-test-mget-129d3802

mjob-simple.test.js
✔ mjob create --close -or "echo hello"

mmpu.test.js
✖ mmpu create C_OBJ_PATH -c 1 -H m-custom-header:foo -s TEXT_SIZE -m TEXT_MD5

VError: exec "/home/dap/node-manta-244/bin/mmpu create /poseidon/stor/node-manta-test-mmpu-poseidon-commit -c 1 -H m-custom-header:foo -s 9 -m 3pNu6QUfS3RORVBpMh20lA==": exited with status 1
    at interpretChildProcessResult (/home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:163:15)
    at onChildExited (/home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:75:9)
    at /home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:53:4
    at ChildProcess.exithandler (child_process.js:662:7)
    at ChildProcess.emit (events.js:98:17)
    at maybeClose (child_process.js:766:16)
    at Process.ChildProcess._handle.onexit (child_process.js:833:5)

✖ mmpu get C_ID

AssertionError: 'object' === '[string]'
    at forkExecWait (/home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:22:17)
    at Object.argv (/home/dap/node-manta-244/test/mmpu.test.js:110:5)
    at Object.<anonymous> (/home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:236:16)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:236:16
    at Object.exports.runTest (/home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:70:9)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:118:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:513:13
    at iterate (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:123:13)
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:134:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:515:17
    at Object._onImmediate (/home/dap/node-manta-244/node_modules/nodeunit/lib/types.js:146:17)
    at processImmediate [as _immediateCallback] (timers.js:363:15)

✖ mmpu create A_OBJ_PATH -c 1 -s TEXT_SIZE -m TEXT_MD5-H durability-level 3 -H content-length:10 -H content-md5:foo

VError: exec "/home/dap/node-manta-244/bin/mmpu create /poseidon/stor/node-manta-test-mmpu-poseidon-abort -c 1 -s 9 -m 3pNu6QUfS3RORVBpMh20lA== -H durability-level:3 -H content-length:10 -H content-md5:foo": exited with status 1
    at interpretChildProcessResult (/home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:163:15)
    at onChildExited (/home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:75:9)
    at /home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:53:4
    at ChildProcess.exithandler (child_process.js:662:7)
    at ChildProcess.emit (events.js:98:17)
    at maybeClose (child_process.js:766:16)
    at Process.ChildProcess._handle.onexit (child_process.js:833:5)

✖ mmpu get A_IDJ

AssertionError: 'object' === '[string]'
    at forkExecWait (/home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:22:17)
    at Object.argv (/home/dap/node-manta-244/test/mmpu.test.js:177:5)
    at Object.<anonymous> (/home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:236:16)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:236:16
    at Object.exports.runTest (/home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:70:9)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:118:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:513:13
    at iterate (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:123:13)
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:134:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:515:17
    at Object._onImmediate (/home/dap/node-manta-244/node_modules/nodeunit/lib/types.js:146:17)
    at processImmediate [as _immediateCallback] (timers.js:363:15)

✖ mmpu parts C_ID: pre-upload

AssertionError: 'object' === '[string]'
    at forkExecWait (/home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:22:17)
    at Object.argv (/home/dap/node-manta-244/test/mmpu.test.js:216:5)
    at Object.<anonymous> (/home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:236:16)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:236:16
    at Object.exports.runTest (/home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:70:9)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:118:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:513:13
    at iterate (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:123:13)
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:134:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:515:17
    at Object._onImmediate (/home/dap/node-manta-244/node_modules/nodeunit/lib/types.js:146:17)
    at processImmediate [as _immediateCallback] (timers.js:363:15)

✖ mmpu parts A_ID: pre-upload

AssertionError: 'object' === '[string]'
    at forkExecWait (/home/dap/node-manta-244/node_modules/forkexec/lib/forkexec.js:22:17)
    at Object.argv (/home/dap/node-manta-244/test/mmpu.test.js:235:5)
    at Object.<anonymous> (/home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:236:16)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:236:16
    at Object.exports.runTest (/home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:70:9)
    at /home/dap/node-manta-244/node_modules/nodeunit/lib/core.js:118:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:513:13
    at iterate (/home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:123:13)
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:134:25
    at /home/dap/node-manta-244/node_modules/nodeunit/deps/async.js:515:17
    at Object._onImmediate (/home/dap/node-manta-244/node_modules/nodeunit/lib/types.js:146:17)
    at processImmediate [as _immediateCallback] (timers.js:363:15)


FAILURES: Undone tests (or their setups/teardowns): 
- mmpu list

To fix this, make sure all tests call test.done()
make: *** [test] Error 1
```

**Comment by davepacheco** [comment]
I also tested using `-c` with an interactive command:

```
$ ./bin/mlogin -c mdb
 * created interactive job -- f0d2de4f-715c-e1ea-bcc0-b2a97a6f4b4d
 * waiting for session...   established
> $C
mdb: no process active
> ::status
> 


 * remote process exited
 * cleaning up resources...
 * session complete
```

**Comment by davepacheco** [comment]
Proposed change: https://cr.joyent.us/2251

**Comment by davepacheco** [comment]
Code review on patchset 1 suggested that this behavior could reasonably be applied to the "-c" mode as well, as in the case where one uses `-c mdb` and, from mdb, shells out to a command that dumps core.  Patchset 2 addresses this.  New output:

```
$ ./bin/mlogin 
 * created interactive job -- 39d64e35-c7b1-46cc-9c35-8517ab41405a
 * waiting for session...   established
poseidon@manta # node -e 'process.abort()'
Abort (core dumped)
poseidon@manta # ls /var/tmp/core
mrcore.node.35315
poseidon@manta # exit


 * remote process exited
 * cleaning up resources...
 * session complete
dap@b44c74d6 node-manta-244 $ 
```

```
$ ./bin/mlogin -c date
 * created interactive job -- 0fe5a829-e52a-4460-e377-93b4e25dc059
 * waiting for session...   established
Fri Jul 21 19:29:55  2017


 * remote process exited
 * cleaning up resources...
 * session complete
dap@b44c74d6 node-manta-244 $ 
```

```
$ ./bin/mlogin -c mdb
 * created interactive job -- 7ff36bbc-231c-c10c-cfbe-8a2b33b39de4
 * waiting for session...   established
> !node -e 'process.abort()'
> !ls -l /var/tmp/core
total 42139
-rw------- 1 root root 43000741 Jul 21 19:30 mrcore.node.36255
> 


 * remote process exited
 * cleaning up resources...
 * session complete
dap@b44c74d6 node-manta-244 $
```

The `make prepush` output appears to be the same.

**Comment by davepacheco** [comment]
Published as v4.4.3.
